Change the default value of standard_conforming_strings to on.

ahachete opened the pull request:
This change should be publicized to driver maintainers at once and
release-noted as an incompatibility with previous releases.

(cherry picked from commit 0839f312e92c7ab0aecb2c4133197a7da7c5fc39)

Author:             Álvaro Hernández Tortosa <aht@8kdata.com>
Date:               Sat Mar 12 21:20:19 UTC 2016

Changes to doc/src/sgml/{config,syntax}.sgml ignored as they are not
present in gpdb.

ascherbaum-pivotal commented:
@ahachete Did you send in this CLA?

ahachete commented:
Just sent :)


On 21/03/16 22:39, Andreas Scherbaum wrote:
>
> @ahachete <https://github.com/ahachete> Did you send in this CLA?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub 
> <https://github.com/greenplum-db/gpdb/pull/527#issuecomment-199498441>
>

-- 
Álvaro Hernández Tortosa


-----------
8Kdata

ahachete commented:
Should I rebase the patch agains current master?

schubert commented:
CLA is covered (see https://github.com/greenplum-db/gpdb/pull/530). Not sure why the bot didn't pick it up here.

danielgustafsson commented:
Are there any new test failures when running `make installcheck-good` with standards conforming strings turned on?

ahachete commented:
Just rebased with current master (and updated PR). Going to recheck `installcheck-good` tests.

ahachete commented:
The gpcopy tests failed now, as they use the "old" escape syntax. I will need to fix this. I will submit a new patch asap, fixing the test too.

ahachete commented:
I have added a new patch to fix a test within gpcopy that was failing due to the standard_conforming_strings strictness. With this new patch now this branch, which I also rebased with master (as of now) passes the `installcheck-good` tests exactly the same way as master (there are currently 12 tests failing). So I guess it's OK by now. Let me know otherwise.

ascherbaum-pivotal commented:
Just tried this patch on our Pulse environment, and it breaks the "xpath" Bugbuster test.
Can you please look into this?

ahachete commented:
I can't run the bugbuster tests myself. While running on a configured source, where `make installcheck-good` works, well, this is what I get with bugbuster:

```
make installcheck-bugbuster
make: *** No rule to make target `installcheck-bugbuster'.  Stop.
```

Can you please help me run these tests or, alternatively, provide me with the error that happened with `xpath` on your build environment?

andreasscherbaum commented:
There is no Makefile target in the main Makefile :-(

Looks like if you can run the xpath.sql (src/test/regress/bugbuster/sql/xpath.sql) from the psql prompt.

Can you check your Hangout, can send you a tool which helps you setting this up.

ahachete commented:
Hope that this new patch fixes the bugbuster tests too.
Patches have been rebased again current master.

ascherbaum-pivotal commented:
Running Pulse on the updated patch now.

ascherbaum-pivotal commented:
Status update: several Pulse tests failed, but with different errors. Investigation is running.

ahachete commented:
Do those tests fail also on master (without patching)? I'll wait for the investigation results. Thanks!

ascherbaum-pivotal commented:
No, this seems to be a problem with our systems. I'll keep you updated.

ahachete commented:
Oh, OK, thanks.

ascherbaum-pivotal commented:
Issue https://github.com/greenplum-db/gpdb/issues/608 keeps track of the Pulse issues.

ahachete commented:
Commits rebased against current master. How are the Pulse issues going? (checked issue #608 but no further information there). Thanks!

ascherbaum-pivotal commented:
Since the issues with unstable tests in Pulse are going on, but this PR is not related to any of them, I'm going ahead and merge this.

ascherbaum-pivotal commented:
Merged with master.

ascherbaum-pivotal commented:
@ivannovick Who can update the release notes?

danielgustafsson commented:
@mkiyama should probably be involved, there might be places in the main docs that should be updated

ascherbaum-pivotal commented:
Created a Tracker story for @mkiyama, will close this PR now.
